### ISSUE TITLE
Getting Nan values in prediction

### ISSUE THREAD

**Issue by Vincenzo-25** [keras-team/keras#19762]
I am getting nan values when i print the prediccions even when i modified the learning rate, optimizer and neurons. Can someone could tell what to change or improve. Thanks

import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split, RandomizedSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import r2_score, mean_squared_error
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout
from tensorflow.keras.wrappers.scikit_learn import KerasRegressor
from tensorflow.keras.optimizers import Adam, RMSprop

# Cargar los datos
data = pd.read_csv("base de datos python.csv")

# Verificar si hay valores faltantes
if data.isnull().sum().sum() > 0:
    data = data.dropna()

# Separar características y etiquetas
X = data[['este', 'norte', 'cota']]
y = data['ley_au']

# Escalar las características
scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)

# Dividir los datos en conjuntos de entrenamiento y prueba
X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size=0.2, random_state=42)

# Definir la función para crear el modelo Keras
def create_model(neurons=32, hidden_layers=1, dropout_rate=0.2, activation='relu', optimizer='adam'):
    model = Sequential()
    model.add(Dense(neurons, input_dim=X_train.shape[1], activation=activation))
    for _ in range(hidden_layers):
        model.add(Dense(neurons, activation=activation))
        model.add(Dropout(dropout_rate))
    model.add(Dense(1, activation='linear'))
    
    model.compile(loss='mean_squared_error', optimizer=optimizer, metrics=['mean_squared_error'])
    return model

# Crear el modelo KerasRegressor
model = KerasRegressor(build_fn=create_model, verbose=0)

# Definir el espacio de búsqueda de hiperparámetros
param_dist = {
    'neurons': [16, 32, 64, 128],
    'hidden_layers': [1, 2, 3],
    'dropout_rate': [0.1, 0.2, 0.3, 0.4],
    'activation': ['relu', 'tanh'],
    'optimizer': ['adam', 'rmsprop'],
    'batch_size': [16, 32, 64],
    'epochs': [50, 100, 150]
}

# Realizar la búsqueda de hiperparámetros
random_search = RandomizedSearchCV(estimator=model, param_distributions=param_dist, n_iter=20, cv=3, random_state=42, n_jobs=-1)
random_search_result = random_search.fit(X_train, y_train)

# Imprimir los mejores hiperparámetros encontrados
print("Mejores hiperparámetros: ", random_search_result.best_params_)

# Evaluar el modelo en el conjunto de prueba
best_model = random_search_result.best_estimator_
y_pred = best_model.predict(X_test)

# Calcular y mostrar el MSE y R²
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)
print(f"Mean Squared Error en el conjunto de prueba: {mse}")
print(f"R² en el conjunto de prueba: {r2}")

**Comment by SuryanarayanaY** [comment]
Hi @Vincenzo-25 ,

Could you please submit the reproducible code snippet in the form of a colab gist along with dummy dataset?  Nan values can arise if your dataset has also has Nan values or may be due Exploring or vanishing gradients.
